src/ansiblelint/rules/galaxy_collection_version.py

```diff
@@ -0,0 +1,103 @@
+"""Implementation of GalaxyCollectionVersionRule."""
+from __future__ import annotations
+
+import sys
+from functools import total_ordering
+from typing import TYPE_CHECKING, Any
+
+from ansiblelint.errors import MatchError
+from ansiblelint.rules import AnsibleLintRule
+from ansiblelint.utils import LINE_NUMBER_KEY
+
+if TYPE_CHECKING:
+    from typing import Optional
+
+    from ansiblelint.constants import odict
+    from ansiblelint.file_utils import Lintable
+
+
+class GalaxyCollectionVersionRule(AnsibleLintRule):
+    """Rule for checking collection version is greater than 1.0.0"""
+
+    id = "galaxy-collection-version"
+    description = "Confirm via galaxy.yml file if collection version is greater than or equal to 1.0.0"
+    severity = "MEDIUM"
+    tags = ["metadata"]
+    version_added = "v6.5.0 (last update)"
+
+    def matchplay(self, file: Lintable, data: odict[str, Any]) -> list[MatchError]:
+        """Return matches found for a specific play (entry in playbook)."""
+
+        if file.kind != "galaxy":
+            return []
+        if "version" not in data:
+            return [
+                self.create_matcherror(
+                    message="galaxy.yaml should have version tag.",
+                    linenumber=data[LINE_NUMBER_KEY],
+                    tag="collection-version-missing[galaxy]",
+                    filename=file,
+                )
+            ]
+        elif Version(data.get("version")) < Version("1.0.0"):
+            return [
+                self.create_matcherror(
+                    message="collection version should be greater than or equal to 1.0.0",
+                    linenumber=data[LINE_NUMBER_KEY],
+                    tag="collection-version[galaxy]",
+                    filename=file,
+                )
+            ]
+
+
+@total_ordering
+class Version:
```

Comment:
@ssbarnea I wanna move this to somewhere generic, like a utility part for it to be reused among the whole collection. would it be the right candidate for `ansiblelint.utils` ?